src/ethereum_clis/clis/geth.py

```diff
@@ -287,3 +157,207 @@ def _mapping_data(self):
+    A helper class
+    """
+
+    def __init__(self, binary: Path, trace: bool):
+        super().__init__(statetest_binary=binary, blocktest_binary=binary)
+        self.statetest = GethStatetest(binary=binary, trace=trace)
+        self.blocktest = GethBlocktest(binary=binary, trace=trace)
+
+    def consume_fixture(
+        self,
+        fixture_format: FixtureFormat,
+        fixture_path: Path,
+        fixture_name: Optional[str] = None,
+        debug_output_path: Optional[Path] = None,
+    ):
+        """
+        Executes the appropriate geth fixture consumer with the fixture at `fixture_path`.
+        """
+        if fixture_format == StateFixture:
+            return self.statetest.consume(fixture_path, debug_output_path)
+        elif fixture_format == BlockchainFixture:
+            return self.blocktest.consume(fixture_path, fixture_name, debug_output_path)
+        else:
+            raise NotImplementedError(
+                f"Geth can not verify fixture format of type `{fixture_format}`"
+            )
+
+
+class GethEvm(EthereumCLI):
+    """
+    go-ethereum `evm` base class.
+    """
+
+    default_binary = Path("evm")
+    detect_binary_pattern = compile(r"^evm(.exe)? version\b")
+    binary: Path
+    cached_version: Optional[str] = None
+
+    def __init__(self, binary: Path, trace: bool = False, exception_mapper=GethExceptionMapper()):
+        self.binary = binary
+        self.trace = trace
+        self.exception_mapper = exception_mapper
+
+    def _run_command(self, command: List[str]) -> subprocess.CompletedProcess:
+        try:
+            return subprocess.run(
+                command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True
+            )
+        except subprocess.CalledProcessError as e:
+            raise Exception(f"Command failed with non-zero status: {e}.")
+        except Exception as e:
+            raise Exception(f"Unexpected exception calling evm tool: {e}.")
+
+    def _consume_debug_dump(
+        self,
+        command: List[str],
+        result: subprocess.CompletedProcess,
+        fixture_path: Path,
+        debug_output_path: Path,
+    ):
+        debug_fixture_path = debug_output_path / "fixtures.json"
+        consume_direct_call = " ".join(command[:-1]) + f" {debug_fixture_path}"
+        consume_direct_script = textwrap.dedent(
+            f"""\
+            #!/bin/bash
+            {consume_direct_call}
+            """
+        )
+        dump_files_to_directory(
+            str(debug_output_path),
+            {
+                "consume_direct_args.py": command,
+                "consume_direct_returncode.txt": result.returncode,
+                "consume_direct_stdout.txt": result.stdout,
+                "consume_direct_stderr.txt": result.stderr,
+                "consume_direct.sh+x": consume_direct_script,
+            },
+        )
+        shutil.copyfile(fixture_path, debug_fixture_path)
+
+
+class GethTransitionTool(GethEvm, TransitionTool):
```

Comment:
I feel It is not necessarily GethEvm, it just happened to be inside gethEvm
for evmone it will be a standalone class?  it is a bit confusing